themes/default/content/registry/packages/aws/how-to-guides/bucketv2-migration.md

```diff
@@ -0,0 +1,2423 @@
+
+public class App {
+    public static void main(String[] args) {
+        Pulumi.run(App::stack);
+    }
+
+    public static void stack(Context ctx) {
+        final var currentUser = S3Functions.getCanonicalUserId();
+
+        var myBucket = new BucketV2("myBucket");
+
+        var myBucketOwnership = new BucketOwnershipControls("myBucketOwnership", BucketOwnershipControlsArgs.builder()
+            .bucket(myBucket.id())
+            .rule(BucketOwnershipControlsRuleArgs.builder()
+                .objectOwnership("BucketOwnerPreferred")
+                .build())
+            .build());
+
+        var myBucketAcl = new BucketAclV2("myBucketAcl", BucketAclV2Args.builder()
+            .bucket(myBucket.bucket())
+            .accessControlPolicy(BucketAclV2AccessControlPolicyArgs.builder()
+                .owner(BucketAclV2AccessControlPolicyOwnerArgs.builder()
+                    .id(currentUser.applyValue(getCanonicalUserIdResult -> getCanonicalUserIdResult.id()))
+                    .build())
+                .grants(BucketAclV2AccessControlPolicyGrantArgs.builder()
+                    .permission("READ")
+                    .grantee(BucketAclV2AccessControlPolicyGrantGranteeArgs.builder()
+                        .type("CanonicalUser")
+                        .id(currentUser.applyValue(getCanonicalUserIdResult -> getCanonicalUserIdResult.id()))
+                        .build())
+                    .build())
+                .build())
+            .build(), CustomResourceOptions.builder()
+                .dependsOn(myBucketOwnership)
+                .build());
+
+    }
+}
+
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+name: example
+runtime: yaml
+variables:
+  currentUser:
+    fn::invoke:
+      function: aws:s3:getCanonicalUserId
+      arguments: {}
+resources:
+  my-bucket:
+    type: aws:s3:BucketV2
+  my-bucket-ownership:
+    type: aws:s3:BucketOwnershipControls
+    properties:
+      bucket: ${my-bucket.id}
+      rule:
+        objectOwnership: "BucketOwnerPreferred"
+  my-bucket-acl:
+    type: aws:s3:BucketAclV2
+    properties:
+      bucket: ${my-bucket.bucket}
+      accessControlPolicy:
+        owner:
+          id: ${currentUser.id}
+        grants:
+          - permission: "READ"
+            grantee:
+              type: "CanonicalUser"
+              id: ${currentUser.id}
+    options:
+      dependsOn:
+        - ${my-bucket-ownership}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+
+
+
+Note that `dependsOn` on `BucketOwnershipControls` is required for Pulumi to correctly order the operations for this
+infrastructure. Consult the documentation on `aws.s3.BucketAclV2` for more fully worked examples of configuring ACL.
+
+### hostedZoneId input
+
+This will only be available as an output. If your program specified this as an input, simply remove it. The input was
+ignored by prior versions of the provider and was exposed in error.
+
+### other inputs
```

Comment:
How were the props with explicit instructions chosen? I guess they're the most used ones?

I'm wondering if we should have a single example with all options (might not be possible if some settings are mutually exclusive) and show how to migrate that instead.